xbmc/windowing/osx/WinEventsOSXImpl.mm

```diff
@@ -324,19 +322,33 @@ - (NSEvent*)InputEventHandler:(NSEvent*)nsevent
 
 - (XBMC_Event)keyPressEvent:(CGEventRef*)event
 {
-  UniChar unicodeString[MAXUNICODESTRLEN] = {0};
   UniCharCount actualStringLength = 0;
+  // Get stringlength of event
+  CGEventKeyboardGetUnicodeString(*event, 0, &actualStringLength, NULL);
 
-  auto keycode = static_cast<CGKeyCode>(CGEventGetIntegerValueField(*event, kCGKeyboardEventKeycode));
-  CGEventKeyboardGetUnicodeString(*event, MAXUNICODESTRLEN,
+  // Create array with size of event string
+  UniChar unicodeString[actualStringLength];
+  memset(unicodeString, 0, sizeof(unicodeString));
+
+  auto keycode =
+      static_cast<CGKeyCode>(CGEventGetIntegerValueField(*event, kCGKeyboardEventKeycode));
+  CGEventKeyboardGetUnicodeString(*event, sizeof(unicodeString) / sizeof(*unicodeString),
                                   &actualStringLength, unicodeString);
```

Comment:
```suggestion
  CGEventKeyboardGetUnicodeString(*event, actualStringLength,
                                  nullptr, unicodeString);
```

Reply:
The nullptr doesnt actually work. events fail to get processed unless the UniCharCount is provided.